topology.go

```diff
@@ -18,18 +18,17 @@ type Topology struct {
 	Filters []Filter
 	Output  []Output
 	Upload  Upload
+	Metrics MetricsClient
 
 	inerr     atomic.Value
 	inch      chan *Data
 	outch     []chan OutputRecord
 	rawOutput bool
 	upch      chan string
 
-	metrics   MetricsClient
-	malformed int64 // count parse or empty records
-
-	mu      sync.RWMutex         // protects invalid map
-	invalid map[FieldIndex]int64 // tracks validation errors (by field)
+	malformed int64                // count parse or empty records
+	mu        sync.RWMutex         // protects invalid map
+	invalid   map[FieldIndex]int64 // tracks validation errors (by field)
```

Comment:
Leave an empty line between `malformed` and `mu`. It's a convention to visually show what are the field which should be accessed through the mutex

```suggestion
	malformed int64 // count parsing errors and empty records

	mu      sync.RWMutex         // protects invalid map
	invalid map[FieldIndex]int64 // tracks validation errors (by field)
```